feat: configuration files

dr4hcu5-jan opened the pull request:
This pull request integrates the handling of configuration files for the following aspects

- [ ] Database Connection
- [ ] The Movie Database DB
- [ ] HTTP Server

dr4hcu5-jan committed:
chore(deps): adding spf13/viper

dasbenjo reviewed (dismissed):
sieht spannend aus^^

dr4hcu5-jan committed:
chore: creating internal package

dr4hcu5-jan committed:
chore: implementing function calls to read configuration

dr4hcu5-jan committed:
chore(docs): adding docs page for deployment

the page will only contain a todo entry

dr4hcu5-jan committed:
chore(docs): starting documentation for configuration of the backend

dr4hcu5-jan committed:
chore: adding examplatory configuration file

dr4hcu5-jan committed:
chore(deps): adding testify/assert

dr4hcu5-jan committed:
fix: using errors.As insted of errors.Is

dr4hcu5-jan committed:
chore: moving default value for sslmode to constant

dr4hcu5-jan committed:
chore(tests): adding test cases

the test cases are validating the behaviour of the configuration while parsing and in loading the defaults

dr4hcu5-jan committed:
chore(docs): starting documentation about the configuration files

dr4hcu5-jan committed:
chore(tests): emulating executable behavior by changing directories

dr4hcu5-jan committed:
chore: enabling loading of environment variables

dr4hcu5-jan committed:
chore(ci): adding automated tests

dr4hcu5-jan committed:
fix(ci): using correct delimiter for multiline execution of commands
